universal_transfer_operator/src/universal_transfer_operator/data_providers/filesystem/aws/s3.py

```diff
@@ -129,15 +134,15 @@ def download_file(self, file) -> TempFile:
             return TempFile(tmp_file=Path(tmp_file.name), actual_filename=Path(file_name))
 
     @property
-    def verify(self) -> str | bool | None:
-        return self.dataset.extra.get("verify", None)
+    def verify(self) -> Any:
+        return self.dataset.extra.get("verify")
 
     @property
-    def transfer_config_args(self) -> dict | None:
+    def transfer_config_args(self) -> Any:
         return self.dataset.extra.get("transfer_config_args", {})
 
     @property
-    def s3_extra_args(self) -> dict | None:
+    def s3_extra_args(self) -> Any:
         return self.dataset.extra.get("s3_extra_args", {})
```

Comment:
Feeling a bit uneasy about these `Any` return values :see_no_evil: Why didn't it work to use a more specific type?

Reply:
I too don't like it, but since `self.dataset.extra` is of type dict, which is actually dict[Any, Any] we can't be specific here and the user can actually populate any values for the self.dataset.extra.get("s3_extra_args"). 
If we have to narrow it down, we need to change it to `self.dataset.extra` but I'm not sure what all types of values are valid for this at the moment, I'll more that happy to narrow it down if we can figure this out somehow,

Reply:
And this is the reasoning for other methods having `Any`

Reply:
Hm, I see - we'll start like this and review in future :crossed_fingers: